cmake/modules/unittest.cmake

```diff
@@ -33,12 +44,15 @@ if((NOT DEFINED ZEPHYR_BASE) AND (DEFINED ENV_ZEPHYR_BASE))
   set(ZEPHYR_BASE ${ENV_ZEPHYR_BASE} CACHE PATH "Zephyr base")
 endif()
 
+if(SOURCES)
```

Comment:
this if check should be inside the deprecated package, so that code itself is as clean as possible.
So please move the `if(SOURCES)` here:
https://github.com/zephyrproject-rtos/zephyr/blob/442fae1b389c621530a2c379511ea64963d190a8/cmake/modules/FindDeprecated.cmake#L77